File: python/pcube/service_config.py
class ServiceConfig:
    SYSTEM_HOST_QUEUE_NAME = "/mq_queue_host"
    SYSTEM_WORKER_QUEUE_NAME = "/mq_queue_worker"
    def __init__(self, args: list):
        self.is_host = False
        self._match_is_host(args)
        if self.is_host:
            self.q_name_host = ServiceConfig.SYSTEM_HOST_QUEUE_NAME
            self.q_name_worker = ServiceConfig.SYSTEM_WORKER_QUEUE_NAME
        else:
            self.q_name_host = ServiceConfig.SYSTEM_WORKER_QUEUE_NAME
            self.q_name_worker = ServiceConfig.SYSTEM_HOST_QUEUE_NAME

    def _match_is_host(self, args: list):
        for arg in args:
            if arg == "--host":
                self.is_host = True
                args.remove(arg)
                break

File: rust/src/pcube/service_config.rs

const SYSTEM_HOST_QUEUE_NAME: &str = "/mq_queue_host";
const SYSTEM_WORKER_QUEUE_NAME: &str = "/mq_queue_worker";

pub struct ServiceConfig 
{
    pub is_host: bool,
    pub q_name_host: Option<String>,
    pub q_name_worker: Option<String>
}

impl ServiceConfig
{
    pub fn new(args: &mut Vec<String>) -> Self 
    {
        let mut service_config = Self {is_host: false, q_name_host: None, q_name_worker: None};
        service_config.match_is_host(args);
        if service_config.is_host
        {
            service_config.q_name_host = Some(SYSTEM_HOST_QUEUE_NAME.to_string());
            service_config.q_name_worker = Some(SYSTEM_WORKER_QUEUE_NAME.to_string());
        }
        else
        {
            service_config.q_name_host = Some(SYSTEM_WORKER_QUEUE_NAME.to_string());
            service_config.q_name_worker = Some(SYSTEM_HOST_QUEUE_NAME.to_string());
        }
        return service_config;
    }

    pub fn match_is_host(&mut self, args: &mut Vec<String>)
    {
        match args.iter().position(|n| n == "--host")
        {
            Some(position) => 
            {
                self.is_host=true;
                args.remove(position);
            },
            None => {}
        };
    }
}